lib/active_model_serializers/adapter/base.rb

```diff
@@ -54,25 +54,24 @@ def include_meta(json)
         json
       end
 
-      def default_key_transform
-        :unaltered
-      end
+      class << self
+        def default_transform
+          :unaltered
+        end
 
-      # Determines the transform to use in order of precedence:
-      #   serialization context, global config, adapter default.
-      #
-      # @param serialization_context [Object] the SerializationContext
-      # @return [Symbol] the transform to use
-      def key_transform(serialization_context)
-        serialization_context.key_transform ||
-        ActiveModelSerializers.config.key_transform ||
-        default_key_transform
-      end
+        # Determines the transform to use in order of precedence:
+        #   adapter option, global config, adapter default.
+        #
+        # @param options [Object]
+        # @return [Symbol] the transform to use
+        def transform(options)
+          return options[:transform] if options && options[:transform]
+          ActiveModelSerializers.config.transform || default_transform
+        end
 
-      def transform_key_casing!(value, serialization_context)
-        return value unless serialization_context
-        transform = key_transform(serialization_context)
-        KeyTransform.send(transform, value)
+        def transform_casing!(value, options)
```

Comment:
Determines the transform that should be used, then performs the transform on the supplied value.